compact_str/src/repr/boxed/mod.rs

```diff
@@ -694,28 +757,100 @@ mod tests {
         // assert the string is still correct
         assert_eq!(&format!("{}!", string), box_string.as_str());
 
-        // on 32-bit archs the capacity will be stored on the heap
-        #[cfg(target_pointer_width = "32")]
-        assert!(box_string.cap.as_usize().is_err());
-        // on 64-bit archs it's still inlined
-        #[cfg(not(target_pointer_width = "32"))]
-        assert!(box_string.cap.as_usize().is_ok());
+        cfg_if::cfg_if! {
+            if #[cfg(target_pointer_width = "64")] {
+                // on 64-bit archs it's still inlined
+                assert!(box_string.cap.as_usize().is_ok());
+            } else {
+                // on 32-bit archs the capacity will be stored on the heap
+                assert!(box_string.cap.as_usize().is_err());
+            }
+        }
 
         // push a string
         box_string.push_str("hello!");
 
-        // on 32-bit archs the capacity will still be stored on the heap, since the capacity hasn't
-        // changed
-        #[cfg(target_pointer_width = "32")]
-        assert!(box_string.cap.as_usize().is_err());
-        // on 64-bit archs it's still inlined
-        #[cfg(not(target_pointer_width = "32"))]
-        assert!(box_string.cap.as_usize().is_ok());
+        cfg_if::cfg_if! {
+            if #[cfg(target_pointer_width = "64")] {
+                // on 64-bit archs it's still inlined
+                assert!(box_string.cap.as_usize().is_ok());
+            } else {
```

Comment:
I think the `target_pointer_width = "32"` check should stay, and `compile_error!()` if the pointer width is neither. Some seem to work on porting rust to [msp430-none-elf](https://doc.rust-lang.org/nightly/rustc/platform-support.html), which is a 16bit platform.

Reply:
Good idea! I updated all the tests to use `compile_error!()`